web/auth/rate_limiting.go

```diff
@@ -98,7 +98,14 @@ func NewRedisCounter(client redis.UniversalClient) Counter {
 }
 
 func (r *redisCounter) Increment(key string, timeLimit time.Duration) (int64, error) {
-	return r.Client.Incr(key).Result()
+	count, err := r.Client.Incr(key).Result()
+	if err != nil {
+		return 0, err
+	}
+	if count == 1 {
+		r.Client.Expire(key, timeLimit)
```

Comment:
Non blocking : You probably should have used Redis transactions to deal with concurrency. You may have 1000 requests done by another client between L101 and L106 here

Reply:
I don't see the concurrency issue here. Can you explain why it matters if 1000 requests have been made between L101 and L106?

Reply:
It's not an issue (so non blocking) but probably a bad pattern to not use a transaction. No real consequence here

Reply:
And, even if we have a race condition, I don't think we can do what we want with a redis transaction (neither multi/exec nor watch/discard), but there is still the lua script way.

Reply:
> probably a bad pattern to not use a transaction

I disagree: it's true for most databases, but not redis. Redis is single-threaded and its transactions are not comparable to transactions in SQL world for example.